src/sql/azdata.proposed.d.ts

```diff
@@ -976,4 +976,168 @@ declare module 'azdata' {
 			expiresOn?: number
 		}
 	}
+
+	export enum DataProviderType {
+		TableDesignerProvider = 'TableDesignerProvider'
+	}
+
+	export namespace dataprotocol {
+		export function registerTableDesignerProvider(provider: designers.TableDesignerProvider): vscode.Disposable;
+	}
+
+	export namespace designers {
+
+		export function openTableDesigner(providerId: string, tableInfo: TableInfo): Promise<void>;
```

Comment:
During the review you mentioned that you were exposing designers as a generic "object designer" in addition to the table designer shown here. But I'm not seeing that. Are you planning on having that or did I misunderstand something?